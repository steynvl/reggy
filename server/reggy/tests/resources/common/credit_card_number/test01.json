{
  "type": "Credit card number",
  "information": {
    "visa": true,
    "dinersClub": false,
    "masterCard": false,
    "discover": false,
    "americanExpress": false,
    "jcb": false,
    "spacesAndDashes": "No spaces or dashes"
  },
  "generalRegexInfo": {
    "startRegexMatchAt": "Start of line",
    "endRegexMatchAt": "End of line",
    "regexTarget": "Java"
  },
  "generateMethod": "commonUseCases"
}