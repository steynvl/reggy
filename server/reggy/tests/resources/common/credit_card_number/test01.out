Java
Pattern regex = Pattern.compile("^4\\d{12}(?:\\d{3})?$");
^4\\d{12}(?:\\d{3})?$
Perl
my $regex = /^4\d{12}(?:\d{3})?$/;
^4\d{12}(?:\d{3})?$
POSIX
^4\d{12}(?:\d{3})?$
^4\d{12}(?:\d{3})?$
Python
regex = re.compile(r'^4\d{12}(?:\d{3})?$')
^4\d{12}(?:\d{3})?$
JavaScript
const regex = /^4\d{12}(?:\d{3})?$/;
^4\d{12}(?:\d{3})?$
PHP
$regex = '/^4\d{12}(?:\d{3})?$/';
^4\d{12}(?:\d{3})?$
Golang
regex, _ := regexp.Compile("^4\\d{12}(?:\\d{3})?$")
^4\\d{12}(?:\\d{3})?$
Rust
let re = Regex::new(r"^4\d{12}(?:\d{3})?$").unwrap();
^4\d{12}(?:\d{3})?$
C#
var re = new Regex(@"^4\d{12}(?:\d{3})?$");
^4\d{12}(?:\d{3})?$
Scala
val re = "^4\\d{12}(?:\\d{3})?$".r
^4\\d{12}(?:\\d{3})?$
Kotlin
val regex = Regex("^4\\d{12}(?:\\d{3})?$")
^4\\d{12}(?:\\d{3})?$